src/history/hs_rec.c

```diff
@@ -1002,27 +938,63 @@ __hs_fixup_out_of_order_from_pos(WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor,
 }
 
 /*
- * __hs_delete_key_from_pos --
- *     Delete an entire key's worth of data in the history store assuming that the input cursor is
- *     positioned at the beginning of the key range.
+ * __hs_reinsert_key_from_pos --
+ *     Reinsert an entire key's worth of data in the history store assuming that the input cursor is
+ *     positioned at the beginning of the key range. The reinserted values will have 0 start and
+ *     stop timestamps ensure that they are only visible to transaction id based readers.
  */
 static int
-__hs_delete_key_from_pos(
+__hs_reinsert_key_from_pos(
   WT_SESSION_IMPL *session, WT_CURSOR *hs_cursor, uint32_t btree_id, const WT_ITEM *key)
 {
+    WT_CURSOR *insert_cursor;
     WT_CURSOR_BTREE *hs_cbt;
+    WT_DECL_ITEM(hs_value);
     WT_DECL_RET;
+    WT_HS_TIME_POINT start_time_point, stop_time_point;
     WT_ITEM hs_key;
     WT_UPDATE *upd;
-    wt_timestamp_t hs_start_ts;
-    uint64_t hs_counter;
+    wt_timestamp_t durable_timestamp, hs_start_ts, hs_stop_durable_ts;
+    uint64_t hs_counter, hs_insert_counter, hs_upd_type;
     uint32_t hs_btree_id;
     int cmp;
+    const char *open_cursor_cfg[] = {WT_CONFIG_BASE(session, WT_SESSION_open_cursor), NULL};
 
     hs_cbt = (WT_CURSOR_BTREE *)hs_cursor;
     upd = NULL;
+    insert_cursor = NULL;
+
+    /* Allocate buffer for the history store value. */
+    WT_ERR(__wt_scr_alloc(session, 0, &hs_value));
```

Comment:
I don't think this needs to be a scratch buffer.

Reply:
I'll push up a change to save you time.

Reply:
Thanks!

Reply:
e7d36a7